runtime/klee-libc/strcmp.c

```diff
@@ -10,5 +10,5 @@
 int strcmp(const char *a, const char *b) {
   while (*a && *a == *b)
     ++a, ++b;
-  return *a - *b;
+  return *(const unsigned char*)a - *(const unsigned char*)b;
```

Comment:
What's the point of `const` here?

Reply:
I think the whole loop could be shortened to `for (; *a && *a==*b; ++a, ++b);`...

Reply:
Thanks for the comments. 

In general, it's good practice to preserve `const`ness.  Of course, for this small piece of code it doesn't really matter, but since you asked...

There are many ways to write `strcmp` but the one we have is fine, so I see no good reason to rewrite it...  (this also has nothing to do with this PR, and in general we should refrain from shifting the target, as I have argued before).

Anything else?

Reply:
> it's good practice to preserve constness

I agree but here it just adds boilerplate as the dereferenced pointer is only used in a subtraction. But no hard feelings about that one.

> There are many ways to write `strcmp` but the one we have is fine

Sure I'm just not a big fan of the comma operator in arbitrary places.

> Anything else?

The commit message could state sth. like "according to the C standard" as it requires *(both* [to be] *interpreted as unsigned char)* to make the reason for the change more obvious.